### PR TITLE
Refactor testcase structure

### PR DESCRIPTION
**Pull request by MarcelMue**
Towards https://github.com/giantswarm/giantswarm/issues/1595

Some changes to prepare to have multiple testcases in different folders for parallelism.

### CODE REVIEW

File: framework/integration/config_map.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func createConfigMap(configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+func CreateConfigMap(configMap *corev1.ConfigMap, namespace string) (*corev1.ConfigMap, error) {
```
Comment (MarcelMue) [inline framework/integration/config_map.go]: `namespace` needs to be supplied now, I want to reuse these functions for another testcase. 

I made these functions public to access them from the test package, I might move this into another package to get more alignment with `aws-operator` - WDYT?

File: framework/integration/init.go [hunk 2]
```diff
@@ -17,38 +17,13 @@ import (
...
-	namespace = "finalizer-integration-test"
-)
-
 var (
 	err error
 
 	k8sClient kubernetes.Interface
 )
 
-func init() {
```
Comment (MarcelMue) [inline framework/integration/init.go]: Just some reordering after I made functions needed in tests public.

Again might move this into another folder with a package called `setup` to align with aws-operator but not sure if that makes sense in this case.

File: framework/integration/test/basic/basic_test.go [hunk 3]
```diff
@@ -15,15 +17,17 @@ import (
...
-	operatorName := "test-operator"
 	configMapName := "test-cm"
 	expectedFinalizers := []string{
 		"operatorkit.giantswarm.io/test-operator",
 	}
-	operatorkitFramework, err := newFramework(operatorName)
+	operatorName := "test-operator"
+	testNamespace := "finalizer-integration-basic-test"
+
+	integration.MustSetup(testNamespace)
```
Comment (MarcelMue) [inline framework/integration/test/basic/basic_test.go]: I am not using a `TestMain` to do this stuff currently, can someone explain to me why we use `TestMain` if it's just going to be one testcase per Main anyway?

File: .circleci/config.yml [hunk 4]
```diff
@@ -42,9 +42,9 @@ jobs:
       TEST_DIR: "informer/integration"
     <<: *e2eTest
 
-  e2eTestFrameworkIntegration:
+  e2eTestFrameworkIntegrationBasic:
     environment:
-      TEST_DIR: "framework/integration"
+      TEST_DIR: "framework/integration/test/basic"
```
Comment (MarcelMue) [inline .circleci/config.yml]: Just aligning with the new folder structure.

File: framework/integration/config_map.go (hunk 1 above)
Comment (xh3b4sd) [reply framework/integration/config_map.go]: Separate package sounds good to me to not import the root package.

Comment (kopiczko) [reply framework/integration/config_map.go]: Can't you take the namespace from `configMap`?

Comment (MarcelMue) [reply framework/integration/config_map.go]: Yes, I totally forgot that - changed @kopiczko 

I moved the `config_map.go` and `init.go` to a package called `client`. I simply couldn't think of a better name as they both use and initialize the k8s client. Open to other suggestions.

File: framework/integration/client/config_map.go [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
+	createConfigMap, err := k8sClient.CoreV1().ConfigMaps(configMap.Namespace).Create(configMap)
 	if err != nil {
 		return nil, err
 	}
 
 	return createConfigMap, nil
 }
 
-func getConfigMap(name string) (*corev1.ConfigMap, error) {
+func GetConfigMap(name, namespace string) (*corev1.ConfigMap, error) {
```
Comment (kopiczko) [inline framework/integration/client/config_map.go]: FYI In resource URL and client-go namespace goes before name. IMO it's more natural.

File: framework/integration/client/init.go [hunk 6]
```diff
@@ -127,11 +102,32 @@ func mustSetup() {
...
 		panic(err)
 	}
 }
+
+func init() {
+	k8sClient, err = newK8sClient()
+	if err != nil {
+		panic(err)
+	}
+}
```
Comment (kopiczko) [inline framework/integration/client/init.go]: I personally like to see init() on top (even though it's private). It's kind of magical ✨ function.

Comment (MarcelMue) [reply framework/integration/client/init.go]: Done.

File: framework/integration/client/config_map.go (hunk 5 above)
Comment (MarcelMue) [reply framework/integration/client/config_map.go]: I am not sure about this, I find alphabetical more consistent here. Going to leave it like this for now.

**Comment by MarcelMue** [comment]
Got two approvals, going with this now.

**Pull request merged**